src/lib/Dashboard/PaginatedTab.php

```diff
@@ -0,0 +1,118 @@
+<?php
+
+namespace EzSystems\HybridPlatformUi\Dashboard;
+
+use Symfony\Component\Routing\RouterInterface;
+use Symfony\Component\Templating\EngineInterface;
+
+abstract class PaginatedTab implements TabInterface
+{
+    /** @var EngineInterface */
+    protected $templating;
+
+    /** @var RouterInterface */
+    protected $router;
+
+    /** @var string */
+    protected $template;
+
+    /** @var array */
+    protected $parameters = [];
+
+    /**
+     * @param EngineInterface $templating
+     * @param RouterInterface $router
+     */
+    public function __construct(EngineInterface $templating, RouterInterface $router)
+    {
+        $this->templating = $templating;
+        $this->router = $router;
+    }
+
+    /**
+     * @param string $template
+     */
+    public function setTemplate(string $template)
+    {
+        $this->template = $template;
+    }
+
+    /**
+     * @param string $parameter
+     * @param mixed $value
+     */
+    public function setParameter(string $parameter, $value)
+    {
+        $this->parameters[$parameter] = $value;
+    }
+
+    public function setParameters(array $parameters)
+    {
+        foreach ($parameters as $identifier => $parameter) {
+            $this->setParameter($identifier, $parameter);
+        }
+    }
+
+    /**
+     * @return array
+     */
+    public function getParameters(): array
+    {
+        return $this->parameters;
+    }
+
+    /**
+     * @param string $parameter
+     *
+     * @return mixed
+     */
+    public function getParameter(string $parameter)
+    {
+        return $this->parameters[$parameter];
+    }
+
+    /**
+     * @param int $current
+     * @param int $last
+     *
+     * @return array
+     */
+    protected function getPaginatorUrls(int $current, int $last): array
```

Comment:
With `getPageUrl`, this does not seem to be used anywhere yet, so better to not include that just yet (also I find the implementation a bit weird but it's hard to understand everything without an example of use ;))